app/src/common/shared/com/igalia/wolvic/ui/widgets/MediaControlsWidget.java

```diff
@@ -140,9 +139,7 @@ private void updateUI() {
         mBinding.mediaControlSeekBar.setDelegate(new MediaSeekBar.Delegate() {
             @Override
             public void onSeekDragStart() {
-                mPlayOnSeekEnd = mMedia.isPlaying();
                 mBinding.mediaControlSeekLabel.setVisibility(View.VISIBLE);
-                mMedia.pause();
```

Comment:
So the current code actually makes a lot of sense if this method is called correctly. Note that this should be called when we're dragging the progress indicator. It does not make sense to ask the server for every infinitesimal change we do. It's better to pause while dragging and resume when dragging ends, as the code does.

That said, the current behaviour is broken indeed. So I guess the key here is knowing why we're getting SeekDragStart/End called when we don't drag. Maybe it's a bug in Gecko.

Reply:
Yes, but the race didn't happen in our code (nothing related to SeekDragStart/End or drag), I think it's a bug in Gecko. By saying race, I mean it's between the pause/seek/play for Gecko playing video. I have already investigated this a lot, and actually, we rely on the code here to seek video on the actual web page. I have already debugged this before I made this patch and the 'seekto' event did get triggered once for each seek, with the correct time, but still, we can't get the right result.

And plus, since the current code works by seeking a specific time instead of offset, and we will only `notifySeekProgress()` when drag ends, I don't think we are asking the server for every infinitesimal change we do even if the pause gets removed.

https://github.com/Igalia/wolvic/blob/2abd4d7f80ad9049037a90340092832a55c5a623/app/src/main/assets/extensions/fxr_mediasession/main.js#L45-L54

Reply:
OK I've reviewed the code more in deep now. I though that the *Drag* events were coming from gecko but actually they're just a delegate interface we use to track changes in the seek bar widget. The names are a bit misleading here because they suggest that we're performing a drag although it might be a single click (it could be argued that a click is a zero displacement drag but anyway).

So indeed the fix looks correct because we only perform the actual seek when the touch event is finished.